### PR TITLE
(Revert Revert) Optimize Forms by Submission

### PR DESCRIPTION
**Pull request by dannyroberts**
@benrudolph @biyeun, cleaning up after http://manage.dimagi.com/default.asp?252504, reindexing like Ben suggested. (Sorry for overlooking that in the first place, I should have known better and included a reindex in the original fix.)

### CODE REVIEW

File: corehq/pillows/mappings/user_mapping.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 from pillowtop.es_utils import ElasticsearchIndexInfo
 
-USER_INDEX = es_index("hqusers_2017-03-22")
+USER_INDEX = es_index("hqusers_2017-04-24")
```
Comment (dannyroberts) [inline corehq/pillows/mappings/user_mapping.py]: @calellowitz you were the last one to do a user reindex so I'll ask you; is this all that's needed to trigger a reindex? (And is it the preferred way?)

**Comment by dannyroberts** [comment]
@orangejenny buddy

File: corehq/pillows/mappings/user_mapping.py (hunk 1 above)
Comment (calellowitz) [inline corehq/pillows/mappings/user_mapping.py]: if you are not changing the mapping file you should not need to change the index name. i think you can just run the management command `./manage.py ptop_reindexer_v2 user --in-place` which will reindex all the docs without  creating an entirely new index. The other will create a whole new index (and hold up deploy i believe). Let me know if thats not actually what you were going for. If you do proceed with changing the name, I think there are a few other places it appears in the code base (particularly test code) that you will have to change.

**Review by biyeun — APPROVED** [review]

**Comment by biyeun** [comment]
thanks, @dannyroberts!

File: corehq/pillows/mappings/user_mapping.py (hunk 1 above)
Comment (dannyroberts) [inline corehq/pillows/mappings/user_mapping.py]: Yeah ok that seems better. I'll revert that and run the reindex manually on production, and softlayer. And coordinate w/ scale team on NIC

**Comment by dannyroberts** [comment]
Manually reindex production, softlayer, and NIC like Cal suggested

**Pull request merged**